apis/hive/v1/clusterdeployment_types.go

```diff
@@ -526,6 +526,9 @@ const (
 	// HibernatingReasonPowerStatePaused indicates that we can't/won't discover the state of the
 	// cluster's cloud machines because the powerstate-paused annotation is set.
 	HibernatingReasonPowerStatePaused = "PowerStatePaused"
+	// HibernatingReasonClusterDeploymentDeleted indicates that a Cluster Deployment has been deleted
+	// and that the cluster is deprovisioning unless preserveOnDelete is set to true.
+	HibernatingReasonClusterDeploymentDeleted = "ClusterDeploymentDeleted"
```

Comment:
I think for this particular code path - like `Unknown` state - it doesn't differ for Hibernating and Running conditions so I don't see a point of declaring different reasons for them...

Reply:
Same could be said for `PowerStatePaused`.

Reply:
Ah yes - I think during the initial refactor, we decided to have different API definitions for different conditions, but we can rethink that when we refactor this controller again...

Reply:
Good point. I noticed that `"PowerStatePaused"` also differentiated between `ReadyReasonPowerStatePaused ` and `HibernatingReasonPowerStatePaused` . It does seem like a helpful way to confirm that the reason being used is applicable for the condition it's used for, but perhaps it's redundant in these cases.

Reply:
Sometimes we'll typedef a specific string (examples near here) but in this case we haven't. I recall from playing with this that there isn't a great way to DRY it completely. I'm not gonna lose sleep over it.

Reply:
I'm going to keep them as is for now, and either create a separate commit to remove duplicates or wait until the next refactoring. Does that sound ok?